### PR TITLE
Add E2E export data validation test

### PR DESCRIPTION
**Pull request by namadabu**
## Description
Add an e2e test that exports data to a storage account and validates that the exported data is as expected. When running the test locally, it uses the Azure Storage Emulator to export data to.

## Related issues
Addresses [issue [AB#72263](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/72263)].

## Testing
Test passes locally.

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            // Assert both data are equal
+            Assert.True(ValidateDataFromBothSources(dataFromFhirServer, dataFromExport));
+        }
+
+        private bool ValidateDataFromBothSources(Dictionary<string, string> dataFromServer, Dictionary<string, string> dataFromStorageAccount)
+        {
+            bool result = true;
+            if (dataFromStorageAccount.Count != dataFromServer.Count)
+            {
+                Trace.WriteLine($"Count differs. Exported data count: {dataFromStorageAccount.Count} Fhir Server Count: {dataFromServer.Count}");
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs:61]: Use ITestOutputHelper

Comment (namadabu) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ExportDataValidationTests.cs]: Updated to use ITestOutputHelper

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/InProcTestFhirServer.cs [hunk 2]
```diff
@@ -43,6 +43,9 @@ public InProcTestFhirServer(DataStore dataStore, Type startupType)
             configuration["TestAuthEnvironment:FilePath"] = "testauthenvironment.json";
             configuration["FhirServer:Security:Authentication:Authority"] = "https://inprochost";
 
+            // For local development we will use the Azure Storage Emulator for export.
+            configuration["FhirServer:Operations:Export:StorageAccountConnection"] = "UseDevelopmentStorage=true";
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/InProcTestFhirServer.cs:47]: If we can't connect to the DevelopmentStorage emulator locally is there anyway we can include the suggestion for a developer to start it?

Comment (namadabu) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/InProcTestFhirServer.cs:47]: I did think about that, but wasn't able to find a quick and easy way to do it. The storage emulator is used only for this test. I can maybe call it out in the export documentation.

**Pull request merged**